packages/server/src/utils/stream.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  agentStreamEvent,
+  type WorkflowEventData,
+  type WorkflowStream,
+} from "@llamaindex/workflow";
+// DataStream is deprecated, converting to new API
+import {
+  createDataStream,
+  formatDataStreamPart,
+  type DataStreamWriter,
+  type JSONValue,
+} from "ai";
+
+interface StreamCallbacks {
+  onStart?: (streamWriter: DataStreamWriter) => Promise<void> | void;
+  onCompletion?: (
+    streamWriter: DataStreamWriter,
+    fullContent: string,
+  ) => Promise<void> | void;
+  onError?: (
+    error: Error,
+    streamWriter: DataStreamWriter,
+  ) => Promise<void> | void;
+}
+
+/**
+ * Convert a stream of WorkflowEventData to a Response object.
+ * @param stream - The input stream of WorkflowEventData.
+ * @param options - Optional configuration for the response.
+ * @returns A readable stream of data.
+ */
+export function toDataStream(
+  stream: WorkflowStream<WorkflowEventData<unknown>>,
```

Comment:
this might work too, as you're just using for await
```suggestion
  stream: AsyncIterable<WorkflowEventData<unknown>>,
```